code/modules/mob/living/silicon/silicon.dm

```diff
@@ -66,14 +66,16 @@
 /mob/living/silicon/drop_item()
 	return
 
-/mob/living/silicon/emp_act(severity)
+/mob/living/silicon/robot/emp_act(severity)
 	switch(severity)
 		if(1)
 			src.take_organ_damage(20)
 			Stun(8)
+			src.disable_component("comms", 160)

```

Comment:
`src` is implied here, so you don't have to use it; just call the proc and that's it.